### DISCUSSION TITLE
The Kotlin Compiler plugins will deprecate KSP plugin

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#891] [Q&A]
From the announcement, https://github.com/Kotlin/dataframe/discussions/878, I found there is an example using DataFrame compiler plugin, is this means we will not use KSP plugin in the future?

**Comment by koperagen** [comment]
Hi! When compiler plugins API becomes stable, we will recommend to try replacing KSP with it. After a while we will probably deprecate KSP plugin in favor of compiler plugin. Current status: Compiler plugins API is not stable yet, compilation can easily break when you change version of compiler or highlighting can break if you change version of IDE. So it's not ready to be used in a production development.
I think there are no reasons to keep KSP plugin after that? Compiler plugin will provide same functionality for `@DataSchema` interfaces without building the project, and far more advanced tracking of dataframe schemas based on static analysis of expressions.
